File: Practicas/Practica6/Biblioteca.py
class Libro:
    def __init__(self, titulo, autor, isbn):
        self.__titulo = titulo
        self.__autor = autor
        self.__isbn = isbn
        self.__estado = "disponible"

    def get_titulo(self):
        return self.__titulo

    def get_autor(self):
        return self.__autor

    def get_isbn(self):
        return self.__isbn

    def get_estado(self):
        return self.__estado

    def prestar(self):
        if self.__estado == "disponible":
            self.__estado = "prestado"
        else:
            print("El libro ya está prestado.")

    def devolver(self):
        if self.__estado == "prestado":
            self.__estado = "disponible"
        else:
            print("El libro ya está disponible.")

    def __str__(self):
        return f"Libro(titulo='{self.__titulo}', autor='{self.__autor}', isbn='{self.__isbn}', estado='{self.__estado}')"


class Biblioteca:
    def __init__(self):
        self.__libros = []

    def agregar_libro(self, libro):
        self.__libros.append(libro)

    def buscar_libro_por_isbn(self, isbn):
        for libro in self.__libros:
            if libro.get_isbn() == isbn:
                return libro
        return None

    def prestar_libro(self, isbn):
        libro = self.buscar_libro_por_isbn(isbn)
        if libro:
            libro.prestar()
        else:
            print("Libro no encontrado.")

    def devolver_libro(self, isbn):
        libro = self.buscar_libro_por_isbn(isbn)
        if libro:
            libro.devolver()
        else:
            print("Libro no encontrado.")

    def mostrar_libros(self):
        for libro in self.__libros:
            print(libro)


def main():
    biblioteca = Biblioteca()
    biblioteca.agregar_libro(Libro("El Quijote", "Miguel de Cervantes", "12345"))
    biblioteca.agregar_libro(Libro("Cien años de soledad", "Gabriel García Márquez", "67890"))
    biblioteca.agregar_libro(Libro("El Principito", "Antoine de Saint-Exupéry", "54321"))

    while True:
        print("\n--- Menú de Biblioteca ---")
        print("1. Mostrar colección de libros")
        print("2. Prestar un libro")
        print("3. Devolver un libro")
        print("4. Salir")
        opcion = input("Seleccione una opción: ")

        if opcion == '1':
            print("\nColección de libros:")
            biblioteca.mostrar_libros()

        elif opcion == '2':
            isbn_prestar = input("Ingrese el ISBN del libro a prestar: ")
            biblioteca.prestar_libro(isbn_prestar)

        elif opcion == '3':
            isbn_devolver = input("Ingrese el ISBN del libro a devolver: ")
            biblioteca.devolver_libro(isbn_devolver)

        elif opcion == '4':
            print("Saliendo del programa...")
            break

        else:
            print("Opción inválida, intente de nuevo.")


if __name__ == "__main__":
    main()

File: Practicas/Practica6/Biblioteca.java
import java.util.ArrayList;
import java.util.Scanner;

class Libro {
    private String titulo;
    private String autor;
    private String isbn;
    private String estado;

    public Libro(String titulo, String autor, String isbn) {
        this.titulo = titulo;
        this.autor = autor;
        this.isbn = isbn;
        this.estado = "disponible";
    }

    public String getTitulo() { return titulo; }
    public String getAutor() { return autor; }
    public String getIsbn() { return isbn; }
    public String getEstado() { return estado; }

    public void prestar() {
        if (estado.equals("disponible")) {
            estado = "prestado";
        } else {
            System.out.println("El libro ya está prestado.");
        }
    }

    public void devolver() {
        if (estado.equals("prestado")) {
            estado = "disponible";
        } else {
            System.out.println("El libro ya está disponible.");
        }
    }

    @Override
    public String toString() {
        return "Libro{" + "titulo='" + titulo + '\'' + ", autor='" + autor + '\'' + ", isbn='" + isbn + '\'' + ", estado='" + estado + '\'' + '}';
    }
}

class main {
    private ArrayList<Libro> libros;

    public main() {
        this.libros = new ArrayList<>();
    }

    public void agregarLibro(Libro libro) {
        libros.add(libro);
    }

    public Libro buscarLibroPorISBN(String isbn) {
        for (Libro libro : libros) {
            if (libro.getIsbn().equals(isbn)) {
                return libro;
            }
        }
        return null;
    }

    public void prestarLibro(String isbn) {
        Libro libro = buscarLibroPorISBN(isbn);
        if (libro != null) {
            libro.prestar();
        } else {
            System.out.println("Libro no encontrado.");
        }
    }

    public void devolverLibro(String isbn) {
        Libro libro = buscarLibroPorISBN(isbn);
        if (libro != null) {
            libro.devolver();
        } else {
            System.out.println("Libro no encontrado.");
        }
    }

    public void mostrarLibros() {
        for (Libro libro : libros) {
            System.out.println(libro);
        }
    }
}

public class Biblioteca {
    public static void main(String[] args) {
        main biblioteca = new main();
        Scanner scanner = new Scanner(System.in);

        biblioteca.agregarLibro(new Libro("El Quijote", "Miguel de Cervantes", "12345"));
        biblioteca.agregarLibro(new Libro("Cien años de soledad", "Gabriel García Márquez", "67890"));
        biblioteca.agregarLibro(new Libro("El Principito", "Antoine de Saint-Exupéry", "54321"));

        int opcion;
        do {
            System.out.println("\n--- Menú de Biblioteca ---");
            System.out.println("1. Mostrar colección de libros");
            System.out.println("2. Prestar un libro");
            System.out.println("3. Devolver un libro");
            System.out.println("4. Salir");
            System.out.print("Seleccione una opción: ");
            opcion = scanner.nextInt();
            scanner.nextLine(); // Limpiar el buffer

            switch (opcion) {
                case 1:
                    System.out.println("\nColección de libros:");
                    biblioteca.mostrarLibros();
                    break;

                case 2:
                    System.out.print("Ingrese el ISBN del libro a prestar: ");
                    String isbnPrestar = scanner.nextLine();
                    biblioteca.prestarLibro(isbnPrestar);
                    break;

                case 3:
                    System.out.print("Ingrese el ISBN del libro a devolver: ");
                    String isbnDevolver = scanner.nextLine();
                    biblioteca.devolverLibro(isbnDevolver);
                    break;

                case 4:
                    System.out.println("Saliendo del programa...");
                    break;

                default:
                    System.out.println("Opción inválida, intente de nuevo.");
            }
        } while (opcion != 4);

        scanner.close();
    }
}